corehq/apps/api/domain_metadata.py

```diff
@@ -62,7 +76,16 @@ def obj_get_list(self, bundle, **kwargs):
         if kwargs.get('domain'):
             return [self.obj_get(bundle, **kwargs)]
         else:
-            return list(Domain.get_all())
+            filters = {}
+            if hasattr(bundle.request, 'GET'):
+                filters = bundle.request.GET
+            domains = list(Domain.get_all())
```

Comment:
@dannyroberts @czue @fyniac Maybe we can make mapping in sql (like here: https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/casexml/apps/stock/models.py#L95). 

I made some quick version of improved api: https://github.com/dimagi/commcare-hq/compare/master...lwyszomi:api-performance-fix?expand=1

What do you think about this solution? I don't know number of domains on HQ so I'm not sure if it's good idea to move domains ids to sql.